dvc/remote/local.py

```diff
@@ -385,6 +385,10 @@ def _save_file(self, path_info):
         self.link(cache, path)
         self.state.update_link(path)
 
+        # md5 is known, write method to update for known path and md5
+        # self.state.update_state_for_known_path_checksum(path, md5)
+        self.state.update(path, md5)
```

Comment:
Let's also `self.state.update(cache, md5)` here, so that things like `dvc status` don't re-calculate checksums for it. Could you also add a test for that, please? Also the comment for this line is kinda hard to wrap your head around, maybe we could improve it?